src/mca/schizo/ompi/schizo_ompi.c

```diff
@@ -914,6 +917,17 @@ static int setup_fork(prrte_job_t *jdata, prrte_app_context_t *context) {
             prrte_setenv("OMPI_ARGV", argv_buf, true, &app->env);
             free(argv_buf);
         }
+
+        /* Tell each app the arch - if available */
+#ifdef HAVE_SYS_UTSNAME_H
+        struct utsname sysname = {0};
+        if(-1 < uname(&sysname)) {
+            if((sysname.machine[0] != 0) && (sysname.machine[0] != '\0')) {
+                prrte_setenv("OMPI_MCA_prrte_cpu_type", (const char *) &sysname.machine, true, &app->env);
+            }
```

Comment:
Is there an OMPI MCA param of that name? Or are you making a corresponding change over there? It would be nice if we could eliminate the "prrte" in that name as it shouldn't matter who is providing it.

Reply:
Not yet - it's currently orte specific. Once this gets in I'll pr the change in OMPI.

Reply:
Kewl - then let's remove the "prrte" here. Reason I'm asking is that this code will shortly be getting moved to the pmix/pmdl/ompi component so that all RMs will set these values, not just PRRTE. So keeping things generic to OMPI will be helpful - other RMs won't appreciate peppering their app's environments with references to "prrte"

Reply:
Probably should also do the same for the other envars as I gather you slid one past me already 😄

Reply:
hah - ok - will do!

Reply:
@rhc54 done.

I changed OMPI_MCA_prrte_odls_num_procs -> OMPI_MCA_num_procs and OOMPI_MCA_prrte_cpu_type -> MPI_MCA_cpu_type

Reply:
Thanks!!